core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -283,6 +283,44 @@ protected void validateAddedDataFiles(TableMetadata base, Long startingSnapshotI
     }
   }
 
+  /**
+   * Validates that no delete files matching a filter have been added to the table since a starting snapshot.
+   *
+   * @param base table metadata to validate
+   * @param startingSnapshotId id of the snapshot current at the start of the operation
+   * @param conflictDetectionFilter an expression used to find new conflicting delete files
+   * @param caseSensitive whether expression evaluation should be case sensitive
+   */
+  protected void validateAddedDeleteFiles(TableMetadata base, Long startingSnapshotId,
```

Comment:
Could you update this to `validateNoAddedDeleteFiles` or something similar? It isn't clear from the method name that this is checking that there aren't any. We may want to rename `validateAddedDataFiles` as well.

Reply:
I went for `validateNoNewDeletes` to match `validateNoNewDeletesForDataFiles`.